pom.xml

```diff
@@ -751,6 +758,30 @@
                     <autoReleaseAfterClose>true</autoReleaseAfterClose>
                 </configuration>
             </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-checkstyle-plugin</artifactId>
+                <version>${maven.checkstyle.plugin.version}</version>
+                <configuration>
+                    <configLocation>${checkstyle.config.location}</configLocation>
+                    <headerLocation>${checkstyle.header.file}</headerLocation>
+                    <propertyExpansion>samedir=${checkstyle.basedir}</propertyExpansion>
+                </configuration>
+                <dependencies>
+                    <dependency>
+                        <groupId>com.puppycrawl.tools</groupId>
```

Comment:
what does this dependency do?

Reply:
With this, we can override the version of the underlying checkstyle library. Otherwise, it is tied to the default checkstyle which may be buggy(or not stable). 

Doing so will give us the flexibility to use the stable version of checkstyle.

Reply:
Regarding the library itself:
Checkstyle is a tool for checking Java source code for adherence to a Code Standard or set of validation rules (best practices).

More context - https://github.com/checkstyle/checkstyle

Reply:
plugin version is defined as `<version>${maven.checkstyle.plugin.version}</version>` right? my question is do we really need this dependency? I can see the checkstyle getting applies even if I remove this dependency.

Reply:
Removing this will result in checkstyle giving violations even in Java Annotations.
Let's keep it as it is.